crates/decision/src/decision.rs

```diff
@@ -106,11 +160,21 @@ impl Decision {
     /// # Arguments
     ///
     /// * `threshold` - The minimum value required to accept a [`Decision`].
-    pub fn accepted(&self, threshold: f64) -> bool {
+    pub fn is_accepted(&self, threshold: f64) -> bool {
         let p = self.pignistic();
         p.accept >= threshold
     }
 
+    /// Checks that the [`unknown`](Decision::unknown) value is non-zero while
+    /// [`accept`](Decision::accept) and [`restrict`](Decision::restrict) are both zero.
+    pub fn is_unknown(&self) -> bool {
+        self.unknown >= f64::EPSILON
+            && self.accept >= 0.0
+            && self.accept <= f64::EPSILON
+            && self.restrict >= 0.0
+            && self.restrict <= f64::EPSILON
+    }
```

Comment:
I agree. @coderabbitai also missed that validation doesn't prevent these values from entering a state that would cause the proposed implementation to give an incorrect answer. Some of the functions even rely on temporarily creating invalid values before clamping them back to the valid range, so it's important not to make assumptions that just because there's validation logic, everything must be valid. That's not true.